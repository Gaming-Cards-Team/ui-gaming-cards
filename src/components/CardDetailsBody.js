import React from "react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import CardTextFields from "./CardTextFields";
import { LazyLoadImage } from "react-lazy-load-image-component";
import MockGetCardDetail from "./__mocks__/mockGetCardDetail";
import BlankLines from './BlankLines'

const imageStyle = { 
  textAlign: "center",
  padding: "40px 0px",
};
const productTextStyle = {
  padding: "40px 20px",
};

const  CardDetailsBody = ({id}) => {
  const card = MockGetCardDetail;
  return (
    <div>
      <BlankLines />
      <Row>
        <Col lg={2} md={1} sm={12} xs={12}></Col>
        <Col lg={3} md={4} sm={12} xs={12} style={imageStyle}>
          <LazyLoadImage src={card.image} width={"100%"} effect="blur" />
        </Col>
        <Col lg={4} md={4} sm={12} xs={12}>
          <div style={productTextStyle}>
            <CardTextFields
              rarity={card.rarity_name}
              position={card.position}
              type={card.type_name}
              affiliation={card.affiliation_name}
              color={card.faction_code}
              points={card.points}
              uniqueness={card.uniqueness}
              text={card.text}
            />
          </div>
        </Col>
        <Col lg={3} md={3} sm={12} xs={12}></Col>
      </Row>
    </div>
  );
};

export default CardDetailsBody;
