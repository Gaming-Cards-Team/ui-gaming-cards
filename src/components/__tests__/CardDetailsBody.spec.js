import React from 'react';
import CardDetailsBody from '../CardDetailsBody';

import renderer from 'react-test-renderer';

describe('Test Product Detail Body component', () => {
    it('should validate the component', () => {
        const productDetailBody = renderer.create(<CardDetailsBody id='001' />).toJSON();

        expect(productDetailBody).toMatchSnapshot();
    });
});