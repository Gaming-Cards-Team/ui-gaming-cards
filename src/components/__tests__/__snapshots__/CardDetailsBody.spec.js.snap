// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Product Detail Body component should validate the component 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-lg-2 col-md-1 col-sm-12 col-12"
    />
    <div
      className="col-lg-3 col-md-4 col-sm-12 col-12"
      style={
        Object {
          "padding": "40px 0px",
          "textAlign": "center",
        }
      }
    >
      <div>
        <span
          className=" lazy-load-image-background blur"
          style={
            Object {
              "backgroundImage": "",
              "backgroundSize": "",
              "color": "transparent",
              "display": "inline-block",
              "height": undefined,
              "width": "100%",
            }
          }
        >
          <span
            className=""
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
          />
        </span>
      </div>
    </div>
    <div
      className="col-lg-4 col-md-4 col-sm-12 col-12"
    >
      <div
        style={
          Object {
            "padding": "40px 20px",
          }
        }
      >
        <div>
          <span
            style={
              Object {
                "color": "#2d963f",
                "fontSize": "xx-large",
                "fontWeight": "bold",
              }
            }
          >
             
          </span>
          <h4 />
          <div
            style={
              Object {
                "paddingBottom": 15,
                "paddingTop": 10,
              }
            }
          >
            <div
              className="card"
            >
              <div
                className="card-body"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-6"
                    style={
                      Object {
                        "borderRight": "1px solid #a09797e6",
                      }
                    }
                  >
                    <p>
                      <span
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Rarity: 
                      </span>
                    </p>
                    <p>
                      <span
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Card#: 
                      </span>
                    </p>
                    <p>
                      <span
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Card type: 
                      </span>
                    </p>
                    <p>
                      <span
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Affiliation: 
                      </span>
                    </p>
                    <p>
                      <span
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Color: 
                      </span>
                    </p>
                    <p>
                      <span
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Point value(s): 
                      </span>
                       
                    </p>
                    <p>
                      <span
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Uniqueness: 
                      </span>
                    </p>
                  </div>
                  <div
                    className="col-6"
                  >
                    <p
                      style={
                        Object {
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Description: 
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-lg-3 col-md-3 col-sm-12 col-12"
    />
  </div>
</div>
`;
